nebullvm/utils/data.py

```diff
@@ -99,3 +100,26 @@ def split(self, split_pct: float, shuffle: bool = False):
             DataManager([self[i] for i in idx[:n]]),
             DataManager([self[i] for i in idx[n:]]),
         )
+
+
+class PytorchDataset(torch.utils.data.Dataset):
+    def __init__(self, input_data: DataManager):
+        self.data = input_data
+        self.batch_size = input_data[0][0][0].shape[0]
+        return
+
+    def __len__(self):
+        return sum(
+            [
+                d[0].shape[0]
+                for d in [
+                    [inputs for inputs in batch[0]] for batch in self.data
+                ]
+            ]
+        )
```

Comment:
```suggestion
        return sum(
            [
                batch_inputs[0].shape[0]
                for batch_inputs, _ in self.data
            ]  # we assume the batch size being the first dim of the first input tensor
        )
```

Reply:
Yep I agree, it's definitely better in this way